bincache/Cargo.toml

```
@@ -6,17 +6,21 @@ description = "ZitaneLabs binary cache."
 
 [features]
 default = ["implicit-blocking"]
-implicit-blocking = []
-blocking = []
-rt_tokio_1 = ["dep:tokio"]
-rt_async-std_1 = ["dep:async-std"]
+implicit-blocking = ["dep:futures-io", "dep:futures-util", "async-compression/futures-io"]
+blocking = ["dep:futures-io", "dep:futures-util", "async-compression/futures-io"]
+rt_tokio_1 = ["dep:tokio", "async-compression/tokio"]
+rt_async-std_1 = ["dep:async-std", "async-compression/futures-io"]
+comp_zstd = ["async-compression/zstd"]
 
 [dependencies]
 paste = "1.0.12"
 thiserror = "1.0.40"
 async-trait = { version = "0.1" }
 tokio = { version = "1", features = ["rt", "fs", "io-util"], optional = true }
 async-std = { version = "1", optional = true }
+async-compression = {version = "0.4.0"}
+futures-io = { version = "0.3", features = ["std"], optional = true }
+futures-util = { version = "0.3", features = ["io"], optional = true }
```

I don't think we actually need `futures-io`, since we depend on `futures-util` directly.

```suggestion
implicit-blocking = ["dep:futures-util", "async-compression/futures-io"]
blocking = ["dep:futures-util", "async-compression/futures-io"]
rt_tokio_1 = ["dep:tokio", "async-compression/tokio"]
rt_async-std_1 = ["dep:async-std", "async-compression/futures-io"]
comp_zstd = ["async-compression/zstd"]

[dependencies]
paste = "1.0.12"
thiserror = "1.0.40"
async-trait = { version = "0.1" }
tokio = { version = "1", features = ["rt", "fs", "io-util"], optional = true }
async-std = { version = "1", optional = true }
async-compression = {version = "0.4.0"}
futures-util = { version = "0.3", features = ["io"], optional = true }
```